pax-exam-container-carbon/org.wso2.carbon.container/src/main/java/org/wso2/carbon/container/CarbonContainerFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.wso2.carbon.container;
+
+import org.kohsuke.MetaInfServices;
+import org.ops4j.pax.exam.ExamSystem;
+import org.ops4j.pax.exam.TestContainer;
+import org.ops4j.pax.exam.TestContainerException;
+import org.ops4j.pax.exam.TestContainerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.container.options.CarbonDistributionBaseOption;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.ops4j.pax.exam.CoreOptions.maven;
+
+/**
+ * Factory class for the CarbonTestContainer.
+ */
+@MetaInfServices
+public class CarbonContainerFactory implements TestContainerFactory {
+
+    private static final String DEFAULT_DISTRIBUTION = "org.wso2.carbon.test.default.distribution";
+    private static final Logger logger = LoggerFactory.getLogger(TestContainerFactory.class);
+
+    @Override
+    public TestContainer[] create(ExamSystem system) {
+        if (system.getOptions(CarbonDistributionBaseOption.class).length > 1) {
+            logger.warn("Multiple distribution options are specified in the configuration!!!");
+        }
+
+        List<TestContainer> containers = Arrays.asList(system.getOptions(CarbonDistributionBaseOption.class)).stream()
```

Comment:
With streams, the usual way is to have each operation in a new line. Which is more readable.